src/PsfFlux.cc

```diff
@@ -93,22 +93,23 @@ void PsfFluxAlgorithm::measure(afw::table::SourceRecord& measRecord,
         throw LSST_EXCEPT(MeasurementError, NO_GOOD_PIXELS.doc, NO_GOOD_PIXELS.number);
     }
     typedef afw::detection::Psf::Pixel PsfPixel;
-    auto model = fitRegion.getSpans()
-                         ->flatten(psfImage->getArray(), psfImage->getXY0())
-                         .asEigen<Eigen::ArrayXpr>();
-    auto data = fitRegion.getSpans()
-                        ->flatten(exposure.getMaskedImage().getImage()->getArray(), exposure.getXY0())
-                        .asEigen<Eigen::ArrayXpr>();
-    auto variance = fitRegion.getSpans()
-                            ->flatten(exposure.getMaskedImage().getVariance()->getArray(), exposure.getXY0())
-                            .asEigen<Eigen::ArrayXpr>();
-    PsfPixel alpha = model.matrix().squaredNorm();
+    // SpanSet::flatten returns a new ndarray::Array, which must stay in scope
+    // while we use an Eigen::Map view of it
+    auto modelNdArray = fitRegion.getSpans()->flatten(psfImage->getArray(), psfImage->getXY0());
+    auto dataNdArray = fitRegion.getSpans()->flatten(exposure.getMaskedImage().getImage()->getArray(),
+                                                     exposure.getXY0());
+    auto varianceNdArray = fitRegion.getSpans()->flatten(exposure.getMaskedImage().getVariance()->getArray(),
+                                                         exposure.getXY0());
+    auto model = ndarray::asEigenMatrix(modelNdArray);
+    auto data = ndarray::asEigenMatrix(dataNdArray);
+    auto variance = ndarray::asEigenMatrix(varianceNdArray);
+    PsfPixel alpha = model.squaredNorm();
```

Comment:
Why are these `asEigenMatrix` rather than `asEigenArray`? The original code used `asEigen<Eigen::ArrayXpr>`.

Reply:
Because in almost all cases in the old code the array EigenView is used as a matrix by appending `.matrix()`. This eliminates all those calls to `.matrix()` at the expense of adding one call to `.array()`. See the changes below.